### PR TITLE
IEP-1067 Default config auto creation for debug

### PR DESCRIPTION
**Pull request by sigmaaa**
## Description

When a new project is created we automatically adding a debug config for this project now.

Fixes # ([IEP-1067](https://jira.espressif.com:8443/browse/IEP-1067))

## Type of change

Please delete options that are not relevant.
- New feature (non-breaking change which adds functionality)

## How has this been tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- Test A
- Test B

**Test Configuration**:
* ESP-IDF Version:
* OS (Windows,Linux and macOS):

## Dependent components impacted by this PR:

- Component 1
- Component 2

## Checklist
- [ ] PR Self Reviewed
- [ ] Applied Code formatting
- [ ] Added Documentation
- [ ] Added Unit Test
- [ ] Verified on all platforms - Windows,Linux and macOS

## Summary by CodeRabbit

- **New Features**
    - Enhanced the project creation wizard to automatically create and save a new launch configuration for debugging.
    - Improved project generator initialization for more robust project setup.

### CODE REVIEW

**Comment by AndriiFilippov** [comment]
@sigmaaa  hi ! 

Tested under: 
OS - Windows 10 

Steps: 

1) can't delete **Launch Configuration** (technically it is deleted but still displayed in Launch Config box). 

image


2) can't rename **Launch Configuration**  (same - technically it is renamed but still displayed in Launch Config box)
image

**Comment by sigmaaa** [comment]
> @sigmaaa hi !
> 
> Tested under: OS - Windows 10
> 
> Steps:
> 
> 1. can't delete **Launch Configuration** (technically it is deleted but still displayed in Launch Config box).
> 
> image
> 
> 2. can't rename **Launch Configuration**  (same - technically it is renamed but still displayed in Launch Config box)
>    image

Hi @AndriiFilippov, the default launch configuration can't be renamed or deleted. This behaviour is inherited from the eclipse cdt

File: bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/wizard/NewIDFProjectWizard.java [hunk 1]
```diff
@@ -101,19 +112,61 @@ public boolean performFinish()
...
+		try
+		{
+			wizard.getWorkingCopy().doSave();
+		}
+		catch (CoreException e)
+		{
+			Logger.log(e);
+		}
+		wizard.dispose();
+	}
```
[the remark this answers is not in the document]
Comment (sigmaaa) [reply bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/wizard/NewIDFProjectWizard.java:168]: not critical here

**Comment by AndriiFilippov** [comment]
@sigmaaa hi ! 

Tested under: 
OS - Windows 10 
ESP-IDF: v5.1.2

able to build \ debug with default configurations 🟢 
able to edit configs 🟢 
able to create new configs 🟢 
no issue after Eclipse restart \ exit 🟢 


LGTM 👍

**Review by AndriiFilippov — APPROVED** [review]

**Pull request merged**